assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/llava/llava_mlflow_wrapper.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""MLFlow pyfunc wrapper for LLaVA models."""
+
+import io
+import os
+
+import mlflow
+import pandas as pd
+
+from PIL import Image
+from transformers import TextStreamer
+
+from config import MLflowLiterals, MLflowSchemaLiterals
+from utils import process_image
+
+
+class LLaVAMLflowWrapper(mlflow.pyfunc.PythonModel):
+    """MLflow model wrapper for LLaVA models."""
+
+    LLAVA_MPT = "mpt"
+    LLAVA_7B = "7b"
+    LLAVA_13B = "13b"
+    MODEL_VERSIONS = [LLAVA_MPT, LLAVA_7B, LLAVA_13B]
+
+    def __init__(self, task_type: str) -> None:
+        """Construct LLaVA MLflow wrapper object.
+
+        :param task_type: Task type for training or inference.
+        :type task_type: str
+        """
+        super().__init__()
+
+        self._task_type = task_type
+
+    def load_context(self, context: mlflow.pyfunc.PythonModelContext) -> None:
+        """Load a MLflow model with pyfunc.load_model().
+
+        :param context: MLflow context containing artifacts that the model can use for inference
+        :type context: mlflow.pyfunc.PythonModelContext
+        """
+        from llava.conversation import conv_templates
+        from llava.model.builder import load_pretrained_model
+
+        try:
+            # Get the top level model directory (contains model directory itself and potentially auxiliary directory).
+            model_dir = context.artifacts[MLflowLiterals.MODEL_DIR]
+
+            # Infer the model version from the model directory.
+            self._model_version = next(
+                (n for n in self.MODEL_VERSIONS if model_dir.endswith(n)), None
+            )
+
+            # Set the model name, model base and stop string from the model version.
+            if self._model_version == self.LLAVA_MPT:
+                model_name = "LLaVA-Lightning-MPT-7B-preview"
```

Comment:
Nit: Add model_names strings as constants in config.py ?

Reply:
All the constants in this class could indeed be made into official constants, ie in config.py. However, there is little benefit in doing so because a. the code will only run once and b. this repository basically shows how the model was put into the catalog.
Continuing in the direction of full software engineering, we would need to put "Model loaded successfully" and such into a common file etc.